workflow/controller/operator.go

```diff
@@ -1677,6 +1720,42 @@ func (woc *wfOperationCtx) initializeNodeOrMarkError(node *wfv1.NodeStatus, node
 	return woc.initializeNode(nodeName, wfv1.NodeTypeSkipped, templateScope, orgTmpl, boundaryID, wfv1.NodeError, err.Error())
 }
 
+// Creates a node status that's completely initialized and marked as finished
+func (woc *wfOperationCtx) initializeCacheHitNode(nodeName string, nodeType wfv1.NodeType, templateScope string, orgTmpl wfv1.TemplateReferenceHolder, boundaryID string, outputs *wfv1.Outputs, messages ...string) *wfv1.NodeStatus {
+	woc.log.Debugf("Initializing node %s from cache: template: %s, boundaryID: %s", nodeName, common.GetTemplateHolderString(orgTmpl), boundaryID)
+	nodeID := woc.wf.NodeID(nodeName)
+	_, ok := woc.wf.Status.Nodes[nodeID]
+	if ok {
```

Comment:
surely `!ok`?

Reply:
I think it's `ok` because `ok` is only true if the node already exists